lighthouse-core/scripts/pptr-run-devtools.js

```diff
@@ -56,6 +56,54 @@ const argv = /** @type {Awaited<typeof argv_>} */ (argv_);
 /** @type {LH.Config.Json=} */
 const config = argv.config ? JSON.parse(argv.config) : undefined;
 
+/**
+ * @template R [unknown]
+ * @param {puppeteer.CDPSession} session
+ * @param {string|(() => (R|Promise<R>))} fn
+ * @param {Function[]} [deps]
+ * @return {Promise<R>}
+ */
+async function evaluateInSession(session, fn, deps) {
```

Comment:
Unfortunately, I couldn't find a way to create a puppeteer page for the DevTools frontend target, so I created a couple helper functions for running code in the browser.